datastore/__init__.py

```diff
@@ -251,6 +251,30 @@ async def register_inference_result(
         raise Exception("Unhandled Error")
 
 
+async def new_perfect_inference_feeback(cursor, inference_id, user_id, boxes_id) :
+    """
+    Args:
+        cursor: The cursor object to interact with the database.
+        inference_id (str): id of the inference on which feedback is given
+        user_id (str): id of the user giving a feedback
+        boxes_id (str array): array of id of the objects that are correctly identified
+    """
+    try:
+        for object_id in boxes_id:
+            top_inference_id = inference.get_inference_object_top_id(cursor, object_id)
+            inference.set_inference_object_verified_id(cursor, object_id, top_inference_id )
+            inference.set_inference_object_valid(cursor, object_id, True)
+        
+        # Check if inference is fully verified
+        objects = inference.get_objects_by_inference(cursor, inference_id)
+        if all(obj[4] is not None for obj in objects) :
+            inference.set_inference_feedback_user_id(cursor, inference_id, user_id)
+            inference.set_inference_verified(cursor, inference_id, True)
+        
+    except Exception as e:
```

Comment:
Shouldn't we also catch some of the db.queries.inference errors in case there are issues getting the inferences or objects? This would allow users to have better knowledge about what happened